x/gov/module.go

```diff
@@ -120,6 +121,12 @@ func (am AppModule) BeginBlock(ctx sdk.Ctx, req abci.RequestBeginBlock) {
 
 	ActivateAdditionalParametersACL(ctx, am)
 
+	// On this upgrade height, Pocket Core will start clearing the global session cache to prevent any non-deterministic cache consistency issues.
+	// This code will ensure it is cleared on the upgrade height for a fresh start.
+	if am.keeper.GetCodec().IsOnNamedFeatureActivationHeight(ctx.BlockHeight(), codec.ClearUnjailedValSessionKey) {
```

Comment:
I'd probably add some tolerance (+-2 blocks) just to be safe. 

Whatever hit we take in performance during these few blocks in `BeginBlock` isn't worth the lack of testing/infrastructure we have to be confident that no caching issue will break anything.

Reply:
I'm not sure if this is necessary to clear the cache so many times, but regardless added it along with some unit tests.